### ISSUE TITLE
Dynamic progress bar only prints when accessing progress bar

### ISSUE THREAD

**Issue by wolfv** [p-ranav/indicators#44]
The current implementation only prints when indexing into the progress bar and doesn't expose the print_progress method.

This leads to problems when the last access to the progress bar sets e.g. an option to make the postfix text to "Finalize" it's not going to be printed for the last-to-finish bar.

I think the most formally correct way to implement this would be with either a proxy object (instead of returning a real reference from the accessor, or with a `set_option` / `set_progress` method that also takes an index.

What do you think?

**Comment by p-ranav** [comment]
There is a method in the progress bar class called `.mark_as_completed()` that you can call (by indexing) that will print progress one last time. This was written specifically for this purpose where properties like `postfix_text` need to be updated at the end of progress. Would this work for you?

**Comment by wolfv** [comment]
Hmm, I see, yeah that might work. Anyways, I have some ideas on improvements for the dynamic progress bar. I'll try a PR when I have something. 

My goal would be to have something similar to DNF: 

https://user-images.githubusercontent.com/885054/78633663-71f2e400-78a2-11ea-9f51-c9de76c36d1e.png

so that one also sees some global info like curretn total download speed.

**Comment by wolfv** [comment]
There is an additional issue tied to this where the initial start time is not set for the 1st printing, and then the elapsed time is some very long time :)

**Comment by p-ranav** [comment]
> There is an additional issue tied to this where the initial start time is not set for the 1st printing, and then the elapsed time is some very long time :)

Yeah this should definitely be an issue of its own. I've noticed this on `MultiProgress` too.
